Example/Core/Tests/FIRAppTest.m

```diff
@@ -735,6 +736,19 @@ - (void)testMultipleLibraries {
   XCTAssertTrue([[FIRApp firebaseUserAgent] containsString:@"LegalName/1.0.0 LegalName2/2.0.0"]);
 }
 
+- (void)testRegisteringConformingLibrary {
+  Class testClass = [FIRTestClass class];
+  [FIRApp registerInternalLibrary:testClass withName:@"LegalName" withVersion:@"1.0.0"];
```

Comment:
Any chance we could replace "Legal/Illegal" with "Valid/Invalid"? The term "Legal Name" has other connotations and may be confusing